kuma/wiki/tests/test_views_document.py

```diff
@@ -30,6 +98,299 @@ def ce_settings(settings):
     return settings
 
 
+@pytest.fixture
+def authkey(wiki_user):
+    key = Key(user=wiki_user, description='Test Key 1')
+    secret = key.generate_secret()
+    key.save()
+    auth = '%s:%s' % (key.key, secret)
+    header = 'Basic %s' % base64.encodestring(auth)
+    return AuthKey(key=key, header=header)
+
+
+@pytest.mark.parametrize('method', ('GET', 'HEAD'))
+@pytest.mark.parametrize('if_none_match', (None, 'match', 'mismatch'))
+@pytest.mark.parametrize(
+    'section_case',
+    ('no-section', 'section', 'another-section', 'non-existent-section')
+)
+def test_api_safe(client, sectioned_root_doc, section_case, if_none_match,
+                  method):
+    """
+    Test GET & HEAD on wiki.document_api endpoint.
+    """
+    section_id, exp_content = SECTION_CASE_TO_DETAILS[section_case]
+
+    url = sectioned_root_doc.get_absolute_url() + '$api'
+
+    if section_id:
+        url += '?section={}'.format(section_id)
+
+    headers = {}
+    if if_none_match == 'match':
+        headers['HTTP_IF_NONE_MATCH'] = calculate_etag(
+            sectioned_root_doc.get_html(section_id)
+        )
+    elif if_none_match == 'mismatch':
+        headers['HTTP_IF_NONE_MATCH'] = 'ABC'
+
+    response = getattr(client, method.lower())(url, **headers)
+
+    if if_none_match == 'match':
+        exp_content = ''
+        assert response.status_code == 304
+    else:
+        assert response.status_code == 200
+        assert 'etag' in response
+        assert 'x-kuma-revision' in response
+        assert 'last-modified' not in response
+        assert response['etag'] == '"{}"'.format(calculate_etag(exp_content))
+        assert (response['x-kuma-revision'] ==
+                str(sectioned_root_doc.current_revision_id))
+
+    if method == 'GET':
+        assert response.content == exp_content
+
+
+def test_api_put_forbidden_when_no_authkey(client, wiki_user, root_doc):
+    """
+    A PUT to the wiki.document_api endpoint should forbid access without
+    an authkey, even for logged-in users.
+    """
+    url = root_doc.get_absolute_url() + '$api'
+
+    response = client.put(url)
+    assert response.status_code == 403
+
+    # Even logged-in users without an authkey should be forbidden.
+    wiki_user.set_password('password')
+    wiki_user.save()
+
+    assert client.login(username=wiki_user.username, password='password')
```

Comment:
There are already another fixture name `user_client` which the client is already logged in